src/local_neighborhood.py

```diff
@@ -0,0 +1,121 @@
+import pandas as pd
+import warnings
+from src.PRM import PRM
+from pathlib import Path
+from src.util import prepare_volume, run_container
+
+__all__ = ['LocalNeighborhood']
+
+class LocalNeighborhood(PRM):
+    required_inputs = ['network', 'nodes']
+
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in LocalNeighborhood.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+
+        #
+        #
+        #
+        # I am kind of unsure about how to interpret prize set as for local neiborhood
+        if data.contains_node_columns('prize'):
+            #NODEID is always included in the node table
+            node_df = data.request_node_columns(['prize'])
+        elif data.contains_node_columns(['sources','targets']):
```

Comment:
This logic is a good start. When implementing this part, I found it hard to use the `contains_node_columns` and `request_node_columns` functions and easier to work with the `node_table` of the dataframe directly, checking whether it has a prize, sources, and targets column. Then I added the corresponding nodes to a set of included nodes for each of those relevant columns. You do not necessarily have to do it that way and may prefer to keep closer to this style.

Currently, this implementation will not look at the sources or targets if we find prize nodes. We would like to include prize nodes, sources, _or_ targets.